mediator/src/main/kotlin/pl/edu/prz/kod/mediator/domain/RunnerManager.kt

```diff
@@ -98,4 +100,23 @@ class RunnerManager(
         return response.status.successful && lenses.statusResponseLens(response).status == RunnerStatus.READY
     }
 
+    private fun persistResult(context: ExecutionContext, request: CodeRequest, result: ExecuteRequestResult) {
+        when (result) {
+            is ExecuteRequestResult.Success -> runBlocking {
+                logRepository.insertLog(
+                    ExecutionLog(
+                        email = context.email,
+                        language = request.language,
+                        code = request.base64Code,
+                        stdOut = result.codeResponse.stdOut,
+                        stdErr = result.codeResponse.stdErr,
+                        exitCode = result.codeResponse.exitCode
+                    )
+                )
+                logEvent(LogInsertedEvent(context.email))
```

Comment:
Wouldn't it be necessary to check whether inserting log was successful (result of `insertLog` method) and logging `FailedToInsertLogEvent` if it wasn't?